File: codeExamples/week11/spring24_insta_cult-main/client/src/models/Cult.js
import { DatabaseItem } from "./DatabaseItem.js";
import { Profile } from "./Profile.js";

export class Cult extends DatabaseItem {
  constructor(data) {
    super(data) //calls constructor on DataBaseItem

    // id, createdAt, updatedAt inherited from DatabaseItem

    this.name = data.name
    this.fee = data.fee
    this.description = data.description
    this.coverImg = data.coverImg
    this.leaderId = data.leaderId
    this.leader = new Profile(data.leader)
  }
}

File: codeExamples/week11/spring24_insta_cult-main/server/Models/Cult.cs
namespace insta_cult.Models;


public class Cult : RepoItem<int>
{
  public string Name { get; set; }
  public int Fee { get; set; }
  public string Description { get; set; }
  public string CoverImg { get; set; }
  public string LeaderId { get; set; }
  public Profile Leader { get; set; }
}
